File: frontend/src/types/resource.ts
export interface ResourceResponse {
  id: number;
  name: string;
  type: "RESCUE_TEAM" | "MEDICAL_TEAM" | "BOAT" | "VEHICLE" | "HELICOPTER" | "RELIEF_KIT" | "FOOD_SUPPLY" | "WATER_SUPPLY";
  status: "AVAILABLE" | "DEPLOYED" | "MAINTENANCE" | "UNAVAILABLE";
  current_location: string;
  district: string;
  capacity?: number;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface ResourceAllocationResponse {
  id: number;
  incident_id: number;
  resource_id: number;
  approval_id: number;
  quantity: number;
  status: "ALLOCATED" | "EN_ROUTE" | "ON_SITE" | "RETURNED" | "CANCELLED";
  allocated_at: string;
  dispatched_at?: string;
  completed_at?: string;
}

File: backend/schemas/resource.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from backend.core.enums import ResourceType, ResourceStatus, AllocationStatus

class ResourceResponse(BaseModel):
    id: int
    name: str
    type: ResourceType
    status: ResourceStatus
    current_location: str
    district: str
    capacity: Optional[int] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ResourceAllocationResponse(BaseModel):
    id: int
    incident_id: int
    resource_id: int
    approval_id: int
    quantity: int
    status: AllocationStatus
    allocated_at: datetime
    dispatched_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True
